### ISSUE TITLE
Better error message for EntityType not specified in Schema (due to namespace)

### ISSUE THREAD

**Issue by andrewmwells-amazon** [cedar-policy/cedar#73]
### Category

Cedar validation features

### Describe the feature you'd like to request

In this slack thread: https://cedar-policy.slack.com/archives/C0547KH7R19/p1684968294619939

A user found a confusing error when he had a namespaced schema but his entities file didn't use the namespaced type.

### Describe the solution you'd like

We should warn if types aren't declared in the schema and we should suggest a user likely meant a namespaced type if a matching namespaced type is declared in the schema.

### Describe alternatives you've considered

We could keep the current behavior and update documentation to clarify behavior.

### Is this something that you'd be interested in working on?

- [ ] 👋 I may be able to implement this feature request
- [ ] ⚠️ This feature might incur a breaking change

**Comment by cdisselkoen** [comment]
In the slack thread, the user got the error 
```
failed to parse entities from file mini-entities.json: error while deserializing entities: Attribute "color" on Environment::"env-123" shouldn't exist according to the schema:
```
Notably, they didn't get an error that `Environment` isn't even a type declared in the schema.  Arguably they should.  Any objections to adding this check during schema-based parsing of entities?  Namely, that declaring an entity with a type that isn't declared in the schema should be an error?  Then, it can be that error that we improve the message on in the presence of namespaces

**Comment by john-h-kastner-aws** [comment]
We were talking about how schema based parsing should handle action entities in a another thread. It seems to me that action entities shouldn't be rejected when using schema based parsing because the only valid actions are those defined in the schema. Those should be extracted from the schema and not be duplicated elsewhere and parsed with other entities.

The `Action` entity types for various namespaces aren't explicitly defined in the schema, so emitting an error when an entity type is not defined will have this effect, but we should be delibrate with the error message. Claiming that the `Action` entity type does not exist would clearly be incorrect. It does exist, we just think it's incorrect to try to parse it.

**Comment by cdisselkoen** [comment]
> we should be deliberate with the error message

I'll make sure to have a separate, more informative, error message for `Action` entity types (including namespace+`Action`) vs for all other entity types not declared in the schema.

> action entities should be rejected when using schema based parsing

Worth calling out that this would be a breaking change.  I just confirmed via the CLI that currently, we allow providing action entities in schema-based parsing; we even do this in some of our integration tests (see `cedar-integration-tests/tests/example_use_cases_doc/1a.json` and its entities file `cedar-integration-tests/sample-data/sandbox_a/entities.json`).  I'm in favor of the change though.

**Comment by john-h-kastner-aws** [comment]
>  Worth calling out that this would be a breaking change. I just confirmed via the CLI that currently, we allow providing action entities in schema-based parsing; we even do this in some of our integration tests

Good point. At a minimum, we should check that deleting the action and instead of extracting the action from the schema works for our test cases.

**Comment by cdisselkoen** [comment]
Should we reject action entities in non-schema-based entity parsing as well?  Can we even do that?  If we can't do that, it will be strange that an entities file that was accepted without a schema, is suddenly rejected when you add a schema, even though it was in perfect conformance with the schema, just because you'll be required to remove your action entities when you add the schema.

**Comment by john-h-kastner-aws** [comment]
We definitely need to accept action entities when not using a schema.

If we accept actions entities for schema based parsing, and ultimately want schema based parsing to fill the role of entity store validation, then we'd need to check that the parsed actions are the same as the actions in the schema.

**Comment by cdisselkoen** [comment]
This proposal sounds good to me.  For the purposes of this issue (while we don't have partial schemas), I'll constant-fold the definition of "Consistency" to

1. If the action is present in both places, it matches up precisely.
2. If the action is present only in the entities file, error.
3. If the action is present only in the schema, no further restrictions.
